Makefile

```diff
@@ -160,7 +160,7 @@ test-conformance-fast: ## Run clusterctl based conformance test on workload clus
 ## Binaries
 ## --------------------------------------
 .PHONY: binaries
-binaries: managers clusterawsadm ## Builds and installs all binaries
+binaries: managers clusterawsadm compile-e2e ## Builds and installs all binaries
```

Comment:
I think it might be better to remove this and instead update scripts/ci-build.sh to have it `make binaries compile-e2e`. WDYT?

Reply:
could be done that way as well if we want to keep compilation of test separate.
Question is, should `.PHONY: binaries` make sure that all essential items compile or we want to keep that control outside?

Reply:
I would consider `binaries` to cover the executables that are user-facing and not include testing.

Reply:
makes sense. will push the update.